unitTests/ui/FacetTest.ts

```diff
@@ -759,37 +759,143 @@ export function FacetTest() {
+        it removes the coveo-facet-dependent class from the dependent facet`, () => {
+          masterFacet.cmp.selectValue('master value');
+          Simulate.query(test.env);
+
+          expect($$(test.cmp.element).hasClass('coveo-facet-dependent')).toBe(false);
+        });
+
+        it(`when the master facet has one excluded value,
+        it removes the coveo-facet-dependent class from the dependent facet`, () => {
+          masterFacet.cmp.excludeValue('excluded master value');
+          Simulate.query(test.env);
+
+          expect($$(test.cmp.element).hasClass('coveo-facet-dependent')).toBe(false);
         });
 
-        expect($$(test.cmp.element).hasClass('coveo-facet-dependent')).toBe(true);
-        expect($$(masterFacet.cmp.element).hasClass('coveo-facet-dependent')).toBe(false);
+        describe(`given the master and dependent facet both have one selected and one excluded value`, () => {
+          const selectedMasterValue = 'selected master value';
+          const excludedMasterValue = 'excluded master value';
+
+          function triggerStateChangeOnDependentFacet() {
+            rebindDependentFacetStateChangeListeners();
+            $$(test.env.root).trigger('change', { attributes: {} });
+          }
+
+          function rebindDependentFacetStateChangeListeners() {
+            test.cmp.queryStateModel.getEventName = name => name;
+            test.cmp['initQueryStateEvents']();
+          }
+
+          beforeEach(() => {
+            masterFacet.cmp.selectValue(selectedMasterValue);
+            masterFacet.cmp.excludeValue(excludedMasterValue);
+
+            test.cmp.selectValue('selected dependent value');
+            test.cmp.excludeValue('excluded dependent value');
+          });
+
+          it(`when resetting the master facet,
+          it resets the dependent facet`, () => {
+            masterFacet.cmp.reset();
+            triggerStateChangeOnDependentFacet();
+
+            expect(test.cmp.getSelectedValues().length).toBe(0);
+            expect(test.cmp.getExcludedValues().length).toBe(0);
+          });
+
+          it(`when deselecting one of the master facet active options,
+          it does not reset the dependent facet`, () => {
+            masterFacet.cmp.deselectValue(selectedMasterValue);
+            triggerStateChangeOnDependentFacet();
+
+            expect(test.cmp.getSelectedValues().length).toBe(1);
+            expect(test.cmp.getExcludedValues().length).toBe(1);
+          });
+
+          it(`when deselecting both the master facet active options,
+          it resets the dependent facet`, () => {
+            masterFacet.cmp.deselectValue(selectedMasterValue);
+            masterFacet.cmp.deselectValue(excludedMasterValue);
+            triggerStateChangeOnDependentFacet();
+
+            expect(test.cmp.getSelectedValues().length).toBe(0);
+            expect(test.cmp.getExcludedValues().length).toBe(0);
+          });
```

Comment:
Do we want to test this scenario  :
The master facet has two selected values : 
- masterValue1 
- masterValue2

The dependant facet has two selected values : 
- dependantValue1 : (comes from masterValue1 being selected)
- dependantValue2 : (comes from masterValue2  being selected)

When we unselect the masterValue1... 

From our discussion, I think we expect the dependantValue1 to still be shown and selected right?

Reply:
Isn't this scenario similar to the test on line 880?